doc/user/content/sql/create-source/csv-s3.md

```diff
@@ -0,0 +1,81 @@
+---
+title: "CREATE SOURCE: CSV from an S3 bucket"
+description: "Learn how to connect Materialize to an S3 Bucket"
+menu:
+  main:
+    parent: 'create-source'
+---
+{{< experimental >}}The S3 source type{{< /experimental >}}
+
+{{% create-source/intro %}}
+This document details how to connect Materialize to an S3 Bucket that contains text- or
+byte-encoded CSV-serialized objects.
+{{% /create-source/intro %}}
+
+## Syntax
+
+{{< diagram "create-source-s3-csv.svg" >}}
+
+{{% create-source/syntax-details connector="s3" formats="csv" envelopes="append-only" %}}
+
+## Example
+
+We will assume that there is an S3 bucket "analytics" that contains the following keys and
+associated content:
+
+**users/2021/engagement.csv**
+```csv
+9999,active,8 hours
+888,inactive,
+777,active,3 hours
+```
+
+**users/2020/engagement.csv**
+```csv
+9999,active,750 hours
+888,inactive,
+777,active,1002 hours
+```
+
+We can load all these keys with the following command:
+
+```sql
+> CREATE MATERIALIZED SOURCE csv_example (user_id, status, usage)
+  FROM S3 BUCKET 'analytics' OBJECTS FROM SCAN MATCHING '**/*.csv'
+  WITH (region = 'us-east-2')
+  FORMAT CSV WITH 3 COLUMNS;
+```
+
+This creates a source that...
+
+- Scans the entire `analystics` bucket looking for objects that have keys that end with `*.csv`
+- Has three *text* columns: `user_id`, `status`, and `usage` and one automatically-generated
+  *integer* column `mz_record` which reflects the order that materialized first encountered that
+  row in.o
+
+  Lines in any object that do not have three columns will be ignored.
```

Comment:
Will there be a warning in the log? There should be and it's worth calling out here if so

Reply:
👍 this seems like a silent gotcha, we should definitely call it out in the app (ie, docs only is insufficient)

Reply:
Yes, we produce an error-level log, I'll update these docs to mention it.